tests/middleware/conftest.py

```diff
@@ -0,0 +1,83 @@
+from unittest.mock import Mock
```

Comment:
The middleware tests were refactored to improve coverage for the async version by adopting a similar pattern as the pagination tests.